core/src/apps/solana/ui/__init__.py

```diff
@@ -0,0 +1,281 @@
+from typing import TYPE_CHECKING
+
+from trezor.crypto import base58
+
+from apps.common.paths import address_n_to_str
+
+from ..constants import (
+    ADDRESS_READ_ONLY,
+    ADDRESS_RW,
+    ADDRESS_SIG,
+    ADDRESS_SIG_READ_ONLY,
+)
+from ..transaction.instructions import Instruction
+
+if TYPE_CHECKING:
+    from typing import Any
+
+    # `pass` is Needed because otherwise FW build fails due to invalid syntax error because
+    # `from typing ...` imports are commented out during pre-processing and since there's
+    # only a single import under the `if` statement, the statement would be empty which would result
+    # in an invalid syntax error.
+    pass
+
+
+def _format_property(value: Any, type: str) -> str | bytes | None:
+    if type in ("pubkey", "authority"):
+        return base58.encode(value)
+    elif type == "lamports":
+        return f"{value / 10 ** 9} SOL"
+    elif type == "unix_timestamp":
+        from trezor.strings import format_timestamp
+
+        return format_timestamp(value)
+    elif isinstance(value, int):
+        return str(value)
+
+    return value
+
+
+def _format_path(path: list[int]) -> str:
+    from apps.common.paths import HARDENED
+
+    if len(path) < 4:
+        return address_n_to_str(path)
+
+    ACCOUNT_PATH_INDEX = 3
+    account_index = path[ACCOUNT_PATH_INDEX]
+    unhardened_account_index: int = account_index ^ (account_index & HARDENED)
+    return f"#{unhardened_account_index + 1}"
+
+
+async def show_confirm(
+    instruction: Instruction,
+    instructions_count: int,
+    instruction_index: int,
+    signer_path: list[int],
+    signer_public_key: bytes,
+) -> None:
+    from trezor.enums import ButtonRequestType
+    from trezor.ui.layouts import confirm_metadata, confirm_properties
+
+    # assertions for pyright
+    assert instruction.parsed_data is not None
+    assert instruction.parsed_accounts is not None
+    assert instruction.ui_properties is not None
+
+    instruction_title = (
+        f"{instruction_index}/{instructions_count}: {instruction.ui_name}"
+    )
+
+    if instruction.is_deprecated_warning is not None:
+        await confirm_metadata(
+            "confirm_deprecated_warning",
+            instruction_title,
+            instruction.is_deprecated_warning,
+            br_code=ButtonRequestType.Other,
+        )
+
+    for ui_property in instruction.ui_properties:
+        if ui_property.parameter is not None:
+            property_template = instruction.get_property_template(ui_property.parameter)
+            value = instruction.parsed_data[ui_property.parameter]
+            _type = property_template.type
+
+            if _type == "authority":
+                if signer_public_key == value:
```

Comment:
Could be one-line `if X and Y` instead of nested `if`s

Reply:
[Changes](https://github.com/trezor/trezor-firmware/commit/0e1b2f94c0461a1f18f665d6154d683fe97cf0da)